nixos/modules/hardware/video/webcam/ipu6.nix

```
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.hardware.ipu6;
+
+  kernelPackages = config.boot.kernelPackages;
+
+in
+{
+
+  options.hardware.ipu6 = {
+
+    enable = mkEnableOption (lib.mdDoc "ipu6 kernel module");
+
+    platform = mkOption {
+      default = "ipu6";
+      type = types.enum [ "ipu6" "ipu6ep" ];
+      description = lib.mdDoc ''
+        Set platform to enable kernel module for.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.extraModulePackages = [ kernelPackages.ipu6-drivers ];
```

```suggestion
    boot.extraModulePackages = with config.boot.kernelPackages; [
      ipu6-drivers
    ];
```

We can do the scoping here, given that the binding is not reused anywhere else.

---

done